lxd/endpoints/vmsock.go

```diff
@@ -0,0 +1,51 @@
+package endpoints
+
+import (
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"math"
+	"math/rand"
+	"net"
+
+	"golang.org/x/sys/unix"
+
+	"github.com/lxc/lxd/lxd/util"
+	"github.com/lxc/lxd/lxd/vsock"
+	"github.com/lxc/lxd/shared"
+)
+
+func createVMSockListener(cert *shared.CertInfo) (net.Listener, error) {
+	for i := 0; i < 10; i++ {
+		// Get random port between 1024 and 65535.
+		port := 1024 + rand.Int31n(math.MaxUint16-1024)
+
+		listener, err := vsock.Listen(uint32(port))
+		if err != nil {
+			// Try a different port.
+			if errors.Is(err, unix.EADDRINUSE) {
+				continue
+			}
+
+			return nil, err
+		}
+
+		return tls.NewListener(listener, util.ServerTLSConfig(cert)), nil
+	}
+
+	return nil, fmt.Errorf("Failed creating VM sock listener")
```

Comment:
Presumably this error should be "Failed finding free listen port for VM vsock listener"?